plugin/restore/mysql/mysql.go

```diff
@@ -680,6 +680,181 @@ func (r *Restore) GetSortedBinlogFilesMetaOnServer(ctx context.Context) ([]Binlo
 	return sortBinlogFiles(binlogFiles), nil
 }
 
+// GetBinlogCoordinateByTs converts a timestamp to binlog coordinate using local binlog files.
+func (r *Restore) GetBinlogCoordinateByTs(ctx context.Context, targetTs int64) (*api.BinlogInfo, error) {
+	binlogFilesLocalSorted, err := GetSortedLocalBinlogFiles(r.binlogDir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read sorted local binlog files, error: %w", err)
+	}
+	if !binlogFilesAreContinuous(binlogFilesLocalSorted) {
+		return nil, fmt.Errorf("local binlog files are not continuous")
+	}
+
+	var binlogFileTarget BinlogFile
+	var found bool
+	for i, file := range binlogFilesLocalSorted {
+		eventTs, err := r.parseFirstLocalBinlogEventTs(ctx, file.Name)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse the local binlog file %q's first binlog event ts, error: %w", file.Name, err)
+		}
+		if eventTs >= targetTs {
+			if i == 0 {
+				return nil, fmt.Errorf("the targetTs %d is before the first event ts %d", targetTs, eventTs)
+			}
+			// The previous local binlog file contains targetTs.
+			found = true
+			binlogFileTarget = binlogFilesLocalSorted[i-1]
+			break
+		}
+	}
+	// All of the local binlog files' first event start ts <= targetTs, so we choose the last binlog file as "containing" targetTs.
```

Comment:
nit: this comment doesn't read well.

Reply:
If the binlog file range is [t + 10, t + 20],  I thought specifying t + 25 would throw a similar error as specifying t + 5.

I am not sure secretly moving the restore point is what the user expects

Reply:
SG. Will refactor this.